public/app/plugins/panel/histogram/Histogram.tsx

```diff
@@ -118,20 +118,13 @@ const prepConfig = (frame: DataFrame, theme: GrafanaTheme2) => {
             wantedMax = xScaleMax;
           }
 
-          let fullRangeMin = u.data[0][0];
           let fullRangeMax = u.data[0][u.data[0].length - 1];
 
-          // snap to bucket divisors...
-
-          if (wantedMax === fullRangeMax) {
-            wantedMax += bucketSize;
-          } else {
-            wantedMax = incrRoundUp(wantedMax, bucketSize);
-          }
-
-          if (wantedMin > fullRangeMin) {
-            wantedMin = incrRoundDn(wantedMin, bucketSize);
-          }
+          // isOrdinalX is when we have classic histograms, which are LE, ordinal X, and already have 0 dummy bucket prepended
+          // else we have calculated histograms which are GE and cardinal+linear X, and have no next dummy bucket appended
+          wantedMin = incrRoundUp(wantedMin, bucketSize);
+          wantedMax =
+            !isOrdinalX && wantedMax === fullRangeMax ? wantedMax + bucketSize : incrRoundDn(wantedMax, bucketSize);
```

Comment:
This might be good to break out into a util func that we can add a few basic unit tests to? 🤔

Reply:
this logic may change in the near future and need to be redone in a way that makes such granular tests not worthwhile. we should start visual regression testing of gdev dashboards, so we dont have to write unit tests that check implementation rather than the effect.